lib/libze/libze.c

```diff
@@ -2036,6 +2036,52 @@ typedef struct libze_destroy_cbdata {
     libze_destroy_options *options;
 } libze_destroy_cbdata;
 
+
+typedef struct dependents_cbdata {
+    libze_destroy_cbdata *cbd;
+    zfs_handle_t	*cb_target;
+    nvlist_t	*cb_nvl;
+} dependents_cbdata_t;
+
+
+static int
+promote_dependent_datasets(dependents_cbdata_t *cb)
+{
+    nvpair_t *pair;
+    char *name;
+    for (pair = nvlist_next_nvpair(cb->cb_nvl, NULL);
```

Comment:
I realize this style is common in middleware, however in C, you may want to do something akin to this:

```
pair = nvlist_next_nvpair(cb->cb_nvl, NULL);
for ( ; pair != NULL; pair = nvlist_next_nvpair(cb->cb_nvl, pair)) {
    /*
     * Body of for loop
     */
}
```

or, if the coding convention doesn't mandate an 80 column limit...

```
for (pair = nvlist_next_nvpair(cb->cb_nvl, NULL); pair != NULL; pair = nvlist_next_nvpair(cb->cb_nvl, pair)) {
    /*
     * Body of for loop - Assuming no 80 char wide limit
     */
}
```

... this is **_much_** easier on the eyes